docs/current_docs/api/functions.mdx

```diff
@@ -0,0 +1,209 @@
+
+When a Dagger module is loaded into a Dagger session, the Dagger API is [dynamically extended](./internals.mdx#api-extension-with-dagger-functions) with new functions served by that module. So, after loading a Dagger module, an API client can now call all of the original core functions plus the new functions provided by that module.
+
+## Initialize a Dagger module
+
+Dagger Functions are packaged, shared and reused using Dagger modules. A new Dagger module is initialized by calling `dagger init`. This creates a new `dagger.json` configuration file in the current working directory, together with sample Dagger Function source code. The configuration file will default the name of the module to the current directory name, unless an alternative is specified with the `--name` argument.
+
+Once a module is initialized, `dagger develop --sdk=...` sets up or updates all the resources needed to develop the module locally. By default, the module source code will be stored in the current working directory, unless an alternative is specified with the `--source` argument.
+
+Here is an example of initializing a Dagger module:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+```shell
+dagger init --name=my-module
+dagger develop --sdk=go
+```
+</TabItem>
+<TabItem value="Python">
+```shell
+dagger init --name=my-module
+dagger develop --sdk=python
+```
+</TabItem>
+<TabItem value="TypeScript">
+```shell
+dagger init --name=my-module
+dagger develop --sdk=typescript
+```
+</TabItem>
+</Tabs>
+
+:::info
+At any point, running `dagger develop` regenerates the module's code based on dependencies and the current state of the module.
+:::
+
+The default template from `dagger develop` creates the following structure:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```
+.
+├── LICENSE
+├── dagger
+│   ├── dagger.gen.go
+│   ├── go.mod
+│   ├── go.sum
+│   ├── internal
+│   │   ├── dagger
+│   │   ├── querybuilder
+│   │   └── telemetry
+│   └── main.go
+└── dagger.json
+```
+
+In this structure:
+
+- `dagger.json` is the Dagger module configuration file.
+- `go.mod`/`go.sum` manage the Go module and its dependencies.
+- `main.go` is where your Dagger module code goes. It contains sample code to help you get started.
+- `internal` contains automatically-generated types and helpers needed to configure and run the module:
+    - `dagger` contains definitions for the Dagger API that's tied to the currently running Dagger Engine container.
+    - `querybuilder` has utilities for building GraphQL queries (used internally by the `dagger` package).
+    - `telemetry` has utilities for sending Dagger Engine telemetry.
+
+</TabItem>
+<TabItem value="Python">
+
+```
+.
+├── LICENSE
+├── dagger
```

Comment:
For all SDKs, the default is to put the source files next to `dagger.json`, instead of in a `dagger` subdir.